Classes/Views/DoubleTappableCell.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  GestureCell.swift
+//  Freetime
+//
+//  Created by Joe Rocca on 10/30/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import Foundation
+
+protocol DoubleTappableCellDelegate: class {
+    func didDoubleTap(cell: DoubleTappableCell)
+}
+
+class DoubleTappableCell: UICollectionViewCell, UIGestureRecognizerDelegate {
```

Comment:
Spit-balling here, but wouldn't this also be useful as a protocol? 

Such that if we need a SwipeableDoubleTappableCell we can just implement a hypothetical `Swipeable` and `DoubleTappable`? 

```swift
protocol DoubleTappable: AnyObject, UIGestureRecognizerDelegate {
  var doubleTapGestureRecognizer: UITapGestureRecognizer { get set }
  var doubleTapDelegate: DoubleTappableCellDelgate? { get set }
}

extension DoubleTappable {
  func doubleTapAction() {
    doubleTapDelegate?.didDoubleTap(view: self)
  }

  func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherrGestureRecognizer: UIGestureRecognizer) -> Bool {
    return true
  }
}
```

Reply:
Could even get cute with the gesture recognizer (though not sure it's of _that much value_):

```swift
func DoubleTapGestureRecognizer(containingView: DoubleTappable, delegate: DoubleTappableDelegate) -> UIGestureRecognizer {
    let gestureRecognizer = UITapGestureRecognizer()
    gestureRecognizer.numberOfTapsRequired = 2
    gestureRecognizer.delegate = delegate
    gestureRecognizer.addTarget(containingView, action: #selector(containingView.doubleTapAction))
    return gestureRecognizer
}
```

Reply:
@heshamsalman So I can't do this because Swift protocol extensions are invisible to ObjC. 

https://stackoverflow.com/a/40297075/4645381